### ISSUE TITLE
Calculation errors

### ISSUE THREAD

**Issue by babbishkamath** [moneymanagerex/moneymanagerex#5125]
MMEX version:
 - [ ] 1.5.19
 - [ ] 1.5.20
 - [x] 1.5.21
 - [ ] 1.5.22 Beta
 - [ ] Other (please specify)

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

Please describe your issue with details.
Add screenshot or other files if needed.

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

Expected result:

Actual result:

# Additional information

If you have a backtrace for a crash or a warning, paste it here.

Hi @n-stein, I appreciate providing a new column for realized gains. However after updating to 1.5.21 the same entries looked like this-

![2222](https://user-images.githubusercontent.com/111977534/192005931-11d0ea18-a86a-4ed2-9f04-7324d867214a.png)

Just to check, I deleted the stock in question and re entered the transactions. Now the entry looks like this.Now you can see average cost value for this stock also while for others it is showing zero.
![3333](https://user-images.githubusercontent.com/111977534/192006427-faa53712-4a21-4c6d-a14c-8f24fc63c95d.png)

Also I have another issue. I am not able to figure out how to enter stock splits.

**Comment by n-stein** [comment]
Realized Gain/Loss is correct since it is calculated every time the stock panel is shown, but the average cost is only updated when "Ok" is pressed in the share transaction dialog (i.e. an existing transaction is edited or a new transaction is entered). In the course of normal operation this is sufficient, but it doesn't set the avg cost upon migrating from a build <= 1.5.20 to 1.5.21. 

We can force recalculation by calling Model_Translink::UpdateStockValue when the Stock Panel is displayed. The question is do we do this all the time or only when we find PURCHASEPRICE = 0?

**Comment by n-stein** [comment]
Opened PR #5126 to automatically update the Avg Price of existing stock entries the first time you open the investment account if the stock PURCHASEPRICE = 0.

**Comment by babbishkamath** [comment]
> @babbishkamath If you go to the share account for one of the items which shows 0 avg cost and just open and close (by pressing Ok) one of the transactions does it update properly in the portfolio view? I suspect the issue is right now it is only calculating the avg cost whenever a transaction is entered/changed.

_I checked. Yes it does update the avg cost. But by doing so purchase transactions goto unreconciled mode while sale transactions do not.

**Comment by babbishkamath** [comment]
> Opened PR #5126 to automatically update the Avg Price of existing stock entries the first time you view the investment account if the stock PURCHASEPRICE = 0.

_ I appreciate. I think it is right.

**Comment by babbishkamath** [comment]
> 

But I think I can live with that

**Comment by babbishkamath** [comment]
> > @babbishkamath If you go to the share account for one of the items which shows 0 avg cost and just open and close (by pressing Ok) one of the transactions does it update properly in the portfolio view? I suspect the issue is right now it is only calculating the avg cost whenever a transaction is entered/changed.
> 
> _I checked. Yes it does update the avg cost. But by doing so purchase transactions goto unreconciled mode while sale transactions do not.

I think I can live with that

**Comment by n-stein** [comment]
@babbishkamath Try the beta build of 1.5.22 which has this change incorporated, see if it works as expected.

Win32 portable beta build here:
https://ci.appveyor.com/api/buildjobs/bxvb75hj11ayqbxt/artifacts/mmex-1.5.22-Beta.1-win32-portable.zip

**Comment by babbishkamath** [comment]
> @babbishkamath Try the beta build of 1.5.22 which has this change incorporated, see if it works as expected.
> 
> Win32 portable beta build here: https://ci.appveyor.com/api/buildjobs/bxvb75hj11ayqbxt/artifacts/mmex-1.5.22-Beta.1-win32-portable.zip

Hi @n-stein, Yes avgPrice auto update works as expected, congrats.
 However you may like to check the totalling Loss/Gain
![123](https://user-images.githubusercontent.com/111977534/192130244-e32e8125-4418-4158-ae07-c6be6f5305f8.png)

**Comment by babbishkamath** [comment]
> @babbishkamath Try the beta build of 1.5.22 which has this change incorporated, see if it works as expected.
> 
> Win32 portable beta build here: https://ci.appveyor.com/api/buildjobs/bxvb75hj11ayqbxt/artifacts/mmex-1.5.22-Beta.1-win32-portable.zip

Hi @n-stein, Yes avgPrice auto update works as expected, congrats.
